src/MongoDB.Driver.Core/Core/Operations/EndTransactionOperation.cs

```diff
@@ -64,6 +76,9 @@ public MessageEncoderSettings MessageEncoderSettings
         /// </value>
         public WriteConcern WriteConcern => _writeConcern;
 
+        // internal properties
+        internal BsonDocument RecoveryToken => _recoveryToken;
```

Comment:
I think internal properties and methods are bad. (Internal classes are fine).

Since this is only used for testing let's use reflection instead (unless you think this property should be public).

A possible argument for making it public might be: if `WriteConcern` is public why isn't `RecoveryToken`?

Reply:
Hm fair enough.  I could go either way on this property, but I've removed the property and am using reflection to test it.